src/cargo/sources/git/oxide.rs

```diff
@@ -186,27 +186,102 @@ fn amend_authentication_hints(
     }
 }
 
+/// The reason we are opening a git repository.
+///
+/// This can affect the way we open it and the cost associated with it.
+pub enum OpenMode {
+    /// We need `git_binary` configuration as well for being able to see credential helpers
+    /// that are configured with the `git` installation itself.
+    /// However, this is slow on windows (~150ms) and most people won't need it as they use the
+    /// standard index which won't ever need authentication, so we only enable this when needed.
```

Comment:
I feel like this comment is needed at the use of `git_binary` so when someone adds a new variant, they have the background where its relevant for deciding if the new variant needs to be `true` or `false`

Reply:
I moved the knowledge about the `git_binary` into `OpenMode` in ce08e8b3deb67706f998c70714764851504967f1
 which brings the documentation much closer to each other.

Two birds, one stone, or so I hope :). Does that work?